### PR TITLE
Refactor bridge_info.json to support HA

### PR DESCRIPTION
**Pull request by 2dvorak**
This PR follows up #4 and #5.

The original structure of `bridge_info.json` is as follows:
```
{
    "child" : {
        "url": "http://127.0.0.1:8551",
        "key": "0xc544b44c1c58955af516c1f2ff17f8fd522604f1ea6b64db79e067343ed5e307",
        "operator": "0x7920de6f6d9d1e9fe2e35b48ab8cfce545673888"
    },
    "parent" : {
        "url": "http://127.0.0.1:9781",
        "key": "0x4b07ca7412ad2bb0e62db30369b9f08a8724fb81fce4b3b1af23800233074fbf",
        "operator": "0xa36a4d6d15fa692eceaa616008bc2decc5c0403b"
    }
}
```

This PR suggests to change `bridge_info.json` to:
```
{
  "sender" : {
    "child": {
      "key": "0xc544b44c1c58955af516c1f2ff17f8fd522604f1ea6b64db79e067343ed5e307"
    },
    "parent":
      "key": "0x4b07ca7412ad2bb0e62db30369b9f08a8724fb81fce4b3b1af23800233074fbf"
    }
  }
  "bridges": [
    {
      "child": {
        "url": "http://127.0.0.1:8551",
        "operator": "0xaaaa"
      },
      "parent": {
        "url": "http://127.0.0.1:8552",
        "operator": "0xbbbb"
      }
    },
    {
      "child": {
        "url": "http://127.0.0.1:8553",
        "operator": "0xcccc"
      },
      "parent": {
        "operator": "0xdddd"
      }
    },
  ]
}
```
The `bridges` is a list of bridge URL and operator of parent and child, however note that we need only one parent node to instantiate Caver and send transactions to parent chain. So only the first element of bridge list should have the URL of parent, and the rest can omit that field as shown in above.

The new structure works well even if there is only one bridge; the user can provide only one item in the list.

### CODE REVIEW

File: erc20/alias-erc20-deploy.js [hunk 1]
```diff
@@ -54,48 +54,48 @@ async function deploy(info) {
...
-
-  log = 'register token to subbridge..'
-  await jsonRpcReq(url, log, 'subbridge_registerTokenByAlias', [alias, conf.child.token, conf.parent.token]);
-
-  /*
-   * Initialize service chain configuration with three logs via interaction with attached console
-  console.log(`subbridge.registerBridgeByAlias("${alias}", "${conf.child.bridge}", "${conf.parent.bridge}")`)
-  console.log(`subbridge.subscribeBridgeByAlias("${alias}")`)
-  console.log(`subbridge.registerTokenByAlias("${alias}", "${conf.child.token}", "${conf.parent.token}")`)
-  */
```
Comment (hyunsooda) [inline erc20/alias-erc20-deploy.js:99]: Which fact did you lead to erasing this code?

File: common/bridge_info.json [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
     },
-    "parent" : {
-        "url": "http://127.0.0.1:9781",
-        "key": "0x4b07ca7412ad2bb0e62db30369b9f08a8724fb81fce4b3b1af23800233074fbf",
-        "operator": "0xd5c0321bafc1d82ccd5cb60c0d943328c9d93318"
-    }
+    "bridges": [
+        {
+            "child" : {
+                "url": "http://127.0.0.1:8551",
```
Comment (hyunsooda) [inline common/bridge_info.json:13]: The `url` property is used to caver initialization. It seems reasonable to locate it in the sender's context. Please clarify why did you move it to `bridges`.

File: erc20/alias-erc20-deploy.js (hunk 1 above)
Comment (2dvorak) [reply erc20/alias-erc20-deploy.js:99]: @hyunsooda It was moved into the loop located a few lines above. If HA configured, we have to call `registerBridge`, `subscribeBridge` and `registerToken` in all configured bridges.

**Comment by hyunsooda** [comment]
Could you remove `operator` property in the configuration file and takes it from the RPC response?

The corresponding RPC request function names are:
```
subbridge_getChildOperatorAddr
subbridge_getParentOperatorAddr
```

File: common/bridge_info.json (hunk 2 above)
Comment (2dvorak) [reply common/bridge_info.json:13]: @hyunsooda They are also used for `subbridge` RPCs. I thought the binding between one bridge's URL and operator was more important, so I decided to group them together.

File: erc20/alias-erc20-deploy.js [hunk 3]
```diff
@@ -54,48 +54,48 @@ async function deploy(info) {
...
+
+    log = 'subscribing bridges to the child node'
+    await jsonRpcReq(url, log, 'subbridge_subscribeBridgeByAlias', [alias]);
+
+    log = 'register token to subbridge..'
+    await jsonRpcReq(url, log, 'subbridge_registerTokenByAlias', [alias, conf.contract.child.token, conf.contract.parent.token]);
+  }
+
+  // setOperatorThreshold
+  await conf.contract.child.newInstanceBridge.methods.setOperatorThreshold(conf.bridges.length).send({ from: conf.sender.child.address, gas: 100000000, value: 0 });
```
Comment (hyunsooda) [inline erc20/alias-erc20-deploy.js:86]: This method enables examples to work in the HA setting. A brief description seems needed what it is for and why it is necessary.

**Comment by 2dvorak** [comment]
> Could you remove `operator` property in the configuration file and takes it from the RPC response?
> 
> The corresponding RPC request function names are:
> 
> ```
> subbridge_getChildOperatorAddr
> subbridge_getParentOperatorAddr
> ```

@hyunsooda Thank you for the suggestion, I tried `subbridge_getChildOperator` but without any luck..
Also thank you for finding the proper RPC.

File: erc20/alias-erc20-deploy.js [hunk 4]
```diff
@@ -54,48 +54,48 @@ async function deploy(info) {
...
+    await jsonRpcReq(url, log, 'subbridge_registerBridgeByAlias', [alias, conf.contract.child.bridge, conf.contract.parent.bridge]);
+
+    log = 'subscribing bridges to the child node'
+    await jsonRpcReq(url, log, 'subbridge_subscribeBridgeByAlias', [alias]);
+
+    log = 'register token to subbridge..'
+    await jsonRpcReq(url, log, 'subbridge_registerTokenByAlias', [alias, conf.contract.child.token, conf.contract.parent.token]);
+  }
+
+  // setOperatorThreshold
```
Comment (hyunsooda) [inline erc20/alias-erc20-deploy.js:85]: // set operator threshold

File: common/bridge_info.json (hunk 2 above)
Comment (2dvorak) [reply common/bridge_info.json:13]: @hyunsooda But as you suggested, we can remove `operator` so I think we should design new structure.

**Review by hyunsooda — CHANGES_REQUESTED** [review]

File: erc20/alias-erc20-deploy.js (hunk 3 above)
Comment (2dvorak) [reply erc20/alias-erc20-deploy.js:86]: @hyunsooda Setting the threshold to the length of bridges is NOT necessary, however I thought it would be helpful to find out if all bridges are working fine - if one of the bridges is malfunctioning, the value transfer won't succeed.
But I agree that I should add clarification on that, thank you.

Comment (hyunsooda) [reply erc20/alias-erc20-deploy.js:86]: ok good thx.

File: common/bridge_info.json (hunk 2 above)
Comment (hyunsooda) [reply common/bridge_info.json:13]: Got it.

File: erc20/alias-erc20-deploy.js (hunk 1 above)
Comment (hyunsooda) [reply erc20/alias-erc20-deploy.js:99]: Gotcha. Could you update the comments inside the loop? Those are a manual way to setup the required initialization steps and are referenced from [doc](https://ko.docs.klaytn.com/node/service-chain/getting-started/value-transfer).

Comment (2dvorak) [reply erc20/alias-erc20-deploy.js:99]: This PR will be closed and this review is resolved in the followed PR #8. Thus marking this review resolved.

File: erc20/alias-erc20-deploy.js (hunk 4 above)
Comment (2dvorak) [reply erc20/alias-erc20-deploy.js:85]: This PR will be closed and this review is resolved in the followed PR #8. Thus marking this review resolved.

**Comment by 2dvorak** [comment]
Closed and followed-up by #8.
